build.gradle.kts

```diff
@@ -276,9 +290,40 @@ tasks.shadowJar {
 			"TweakClass" to "org.spongepowered.asm.launch.MixinTweaker",
 		)
 	}
+	
+	doLast {
+		// re-zip the jar with highest compression level
+		val jarFile = shadowJar.archiveFile.get().asFile
+		val fileContents = mutableMapOf<String, ByteArray>()
```

Comment:
Does this map preserve order? I believe JAR files have stricter requirements around entry order than ZIP files.

Reply:
replaced with a LinkedHashMap